### PR TITLE
Bug/490 gnb 7 node testing

### PR DESCRIPTION
**Pull request by ClaudiaComito**
## Description

This PR fixes assorted bugs that led test_gaussiannb to fail on 7 nodes. The problems were related to how heat functions/methods deal with locally empty tensors and, in one case, 1-element tensors.

Issue/s resolved: #490 

## Changes proposed:

- ht.nonzero() now returns dndarray.DNDarray() instead of factories.array(), as the latter gets stuck when one of the nodes has no data (see #492 - which is NOT solved)
- dndarray.__getitem__ in distributed mode now returns the correct result both with 0-dim input (a scalar), and with an empty tensor as input.
- ht.var() ~of a single element tensor no longer returns nan, like torch, but 0.0 like numpy~ and ht.std() now behave like numpy.var() and numpy.std(), as in: whether the Bessel correction is applied or not, is defined by kwarg `ddof`. **N.B. default is now `ddof=0`, corresponding to setting `unbiased=False` in torch.var()**  Thanks @mtar for pointing me to the right direction.

## Type of change

Remove irrelevant options:
- Bug fix WITH breaking change: ht.var(), ht.std()

## Due Diligence

- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

**Comment by ClaudiaComito** [comment]
> Please check your latest commit. It removes older, already merged pull requests.

It should remove the latest merge master. We shouldn't merge master onto bugfix branches, I did by mistake and took it back.

**Comment by mtar** [comment]
okay, I said nothing :zipper_mouth_face:

**Comment by ClaudiaComito** [comment]
> okay, I said nothing 🤐

:) no worries, it is a bit confusing. Thanks for looking into it by the way

**Comment by mtar** [comment]
It's not the only thing that confuses me :confused:. The file comparison is also a mess :dizzy:  :dizzy_face:

File: heat/core/statistics.py [hunk 1]
```diff
@@ -1222,7 +1222,10 @@ def reduce_vars_elementwise(output_shape_i):
 
         if x.lshape[x.split] != 0:
             mu = torch.mean(x._DNDarray__array, dim=axis)
-            var = torch.var(x._DNDarray__array, dim=axis, unbiased=bessel)
+            if x.lshape[x.split] == 1:
+                var = factories.zeros(output_shape_i, dtype=x.dtype, device=x.device)
+            else:
+                var = torch.var(x._DNDarray__array, dim=axis, unbiased=bessel)
```
Comment (mtar) [inline heat/core/statistics.py]: Have you considered that np.var(ddof=1) <=> and np.var <=> torch.tensor(unbiased=False) respectively? In other words, the numpy uses a biased estimator and pytorch an unbiased one by default.
On a single value
- Pytorch's var returns 0 when it's biased.
- numpy's var returns nan when its biased.

**Comment by coquelin77** [comment]
> > Please check your latest commit. It removes older, already merged pull requests.
> 
> It should remove the latest merge master. We shouldn't merge master onto bugfix branches, I did by mistake and took it back.

@mtar was right, by reverting the commit you would delete all of the changes which are upstream of you. you can see this in the line difference. that is why you have about 3000 lines deleted.

**Comment by ClaudiaComito** [comment]
> > > Please check your latest commit. It removes older, already merged pull requests.
> > 
> > 
> > It should remove the latest merge master. We shouldn't merge master onto bugfix branches, I did by mistake and took it back.
> 
> @mtar was right, by reverting the commit you would delete all of the changes which are upstream of you. you can see this in the line difference. that is why you have about 3000 lines deleted.

Guys, you're absolutely right, by reverting the merge I remove the merge. I'm afraid that's exactly what I wanted.

I'm thinking I should make two different PRs of this, since fixing ht.var() and ht.std() has turned into a breaking change.



> > > Please check your latest commit. It removes older, already merged pull requests.
> > 
> > 
> > It should remove the latest merge master. We shouldn't merge master onto bugfix branches, I did by mistake and took it back.
> 
> @mtar was right, by reverting the commit you would delete all of the changes which are upstream of you. you can see this in the line difference. that is why you have about 3000 lines deleted.

File: heat/core/statistics.py (hunk 1 above)
Comment (ClaudiaComito) [reply heat/core/statistics.py]: You're absolutely correct. Thanks a lot. ht.var() and ht.std() are numpy-compliant now

**Comment by ClaudiaComito** [comment]
Closing this PR and replacing it with slightly less messy #519
